tests/test_debugging.py

```diff
@@ -0,0 +1,93 @@
+        expected = {
+            0: np.array([1, 0, 0, 0]),
+            "very_important_state": np.array([1 / np.sqrt(2), 0, 1 / np.sqrt(2), 0]),
+            2: np.array([1 / np.sqrt(2), 0, 0, 1 / np.sqrt(2)]),
+        }
+
+        assert all(k1 == k2 for k1, k2 in zip(result.keys(), expected.keys()))
+        assert all(np.allclose(v1, v2) for v1, v2 in zip(result.values(), expected.values()))
+
+    def test_default_mixed(self):
+        """Test that multiple snapshots are returned correctly on the density-matrix simulator."""
+        dev = qml.device("default.mixed", wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.Snapshot()
+            qml.Hadamard(wires=0)
+            qml.Snapshot("very_important_state")
+            qml.CNOT(wires=[0, 1])
+            qml.Snapshot()
+            return qml.expval(qml.PauliX(0))
+
+        result = qml.snapshots(circuit)()
+        expected = {
+            0: np.array([[1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]]),
+            "very_important_state": np.array(
+                [[0.5, 0, 0.5, 0], [0, 0, 0, 0], [0.5, 0, 0.5, 0], [0, 0, 0, 0]]
+            ),
+            2: np.array([[0.5, 0, 0, 0.5], [0, 0, 0, 0], [0, 0, 0, 0], [0.5, 0, 0, 0.5]]),
+        }
+
+        assert all(k1 == k2 for k1, k2 in zip(result.keys(), expected.keys()))
+        assert all(np.allclose(v1, v2) for v1, v2 in zip(result.values(), expected.values()))
+
+    def test_default_gaussian(self):
+        """Test that multiple snapshots are returned correctly on the CV simulator."""
+        dev = qml.device("default.gaussian", wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.Snapshot()
+            qml.Displacement(0.5, 0, wires=0)
+            qml.Snapshot("very_important_state")
+            qml.Beamsplitter(0.5, 0.7, wires=[0, 1])
+            qml.Snapshot()
+            return qml.expval(qml.X(0))
+
+        result = qml.snapshots(circuit)()
+        expected = {
+            0: {
+                "cov_matrix": np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]]),
+                "means": np.array([0, 0, 0, 0]),
+            },
+            1: {
+                "cov_matrix": np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]]),
+                "means": np.array([1, 0, 0, 0]),
+            },
+            2: {
+                "cov_matrix": np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]]),
+                "means": np.array([0.87758256, 0.36668488, 0, 0.30885441]),
+            },
+        }
+
+        assert all(k1 == k2 for k1, k2 in zip(result.keys(), expected.keys()))
+        assert all(
+            np.allclose(v1["cov_matrix"], v2["cov_matrix"])
+            for v1, v2 in zip(result.values(), expected.values())
+        )
+        assert all(
+            np.allclose(v1["means"], v2["means"])
+            for v1, v2 in zip(result.values(), expected.values())
```

Comment:
I would add some test for each simultator for there is a call of `qml.snapshot` on a qnode that does not contain Snapshot operation. And also add a test for simulators that does not support the `qml.snapshot`function..

Reply:
Great ideas thank you :)